cns/logger/log.go

```diff
@@ -178,16 +156,97 @@ func ResponseEx(tag string, request interface{}, response interface{}, returnCod
 		msg = fmt.Sprintf("[%s] Code:%s, %+v, %+v.", tag, returnCode.String(), request, response)
 	}
 
-	sendTraceInternal(msg)
+	c.sendTraceInternal(msg)
 }
 
-// Send AI telemetry metric
-func SendMetric(metric aitelemetry.Metric) {
-	if Log.th == nil || Log.DisableMetricLogging {
+func (c *CNSLogger) getOrchestratorAndNodeID() (orch, nodeID string) {
+	c.m.RLock()
+	orch, nodeID = c.Orchestrator, c.NodeID
+	c.m.RUnlock()
+	return
+}
+
+func (c *CNSLogger) sendTraceInternal(msg string) {
+	orch, nodeID := c.getOrchestratorAndNodeID()
+
+	report := aitelemetry.Report{
+		Message: msg,
+		Context: nodeID,
+		CustomDimensions: map[string]string{
+			OrchestratorTypeStr: orch,
+			NodeIDStr:           nodeID,
+		},
```

Comment:
Is it worth storing a map as just a general bag of keys and values like many structured logging tools do? We can set "orchestrator" and "nodeID" by default in there, but have the flexibility to add more as necessary. Unsure what the limitations of AI are though, so it may not be feasible :)

Reply:
If I understand correctly, is the suggestion to add a map of default custom dimensions to CNSLogger? I would agree, and we do something similar in DNC. In this case, these properties are lazily added; there will be a time during CNS runtime when they haven't been set by DNC yet, so they are not quite "default". Ideally, once set they should never change, but I don't want to change behavior too much this time around.